openvino/conditional_compilation/include/openvino/cc/selective_build.h

```diff
@@ -245,6 +261,8 @@ bool match(char const *region, Ctx && ctx, T && val, Case && cs, Cases&&... case
 #define OV_SWITCH(Module, fn, ctx, val, ...)    \
     OV_CC_EXPAND(OV_CC_CAT(OV_CC_SWITCH_, OV_CC_SCOPE_IS_ENABLED(OV_CC_CAT3(Module, _, fn)))(Module, fn, ctx, val))
 
+#define OV_DEFAULT() openvino::cc::internal::default_wrapper()
```

Comment:
I think it would be better if we could provide a default type for default cases.
Example:
    OV_SWITCH(CCTests, TestTemplateClass, n, 3,
    OV_CASE(0, int),
    OV_CASE(1, bool),
    OV_CASE(2, float),
    OV_DEFAULT(double));